assets/components/NotificationList.tsx

```diff
@@ -112,94 +113,97 @@ class NotificationList extends React.Component<IProps, IState> {
                     <i className='icon--alert' onClick={this.toggleDisplay} />
                 </span>
 
-                {(() => {
-                    if (this.state.displayItems !== true) {
-                        return null;
-                    }
+                {this.renderPopup(notificationArePaused)}
+            </div>
+        );
+    }
 
-                    if (this.props.fullUser.notification_schedule != null && this.props.fullUser.notification_schedule.pause_from != '' && this.props.fullUser.notification_schedule.pause_to != '' && notificationArePaused) {
-                        return (
-                            <div className="notif__list dropdown-menu dropdown-menu-right show">
-                                <div className='notif__list__header d-flex'>
-                                    <span className='notif__list__header-headline ms-3'>{gettext('Notifications')}</span>
-                                </div>
-        
-                                <div className='d-flex flex-column gap-2 p-3'>
-                                    <div className='nh-container nh-container__text--alert p-2'>
-                                        {gettext('All notifications are paused until {{date}}', {date: formatDate(this.props.fullUser.notification_schedule.pause_to)})}
-                                    </div>
-        
-                                    <button
-                                        type="button"
-                                        className="nh-button nh-button--small nh-button--tertiary"
-                                        onClick={() => {
-                                            this.props.resumeNotifications();
-                                        }}
-                                    >
-                                        {gettext('Resume all notifications')}
-                                    </button>
+    renderPopup(notificationArePaused: boolean): Exclude<React.ReactNode, undefined> {
```

Comment:
If you want to extract this from render method, can you make it a separtate component instead of adding a class method? Class methods are used if you want to access properties from the class itself(via `this`). If not, pure function is the way to go.